File: source/urfaclient.php
<?php
 
require_once (dirname(__FILE__) . "/URFAClient_Connection.php");
require_once (dirname(__FILE__) . "/URFAClient_Packet.php");
require_once (dirname(__FILE__) . "/URFAClient_Admin.php");
require_once (dirname(__FILE__) . "/URFAClient_User5.php");
 
abstract class URFAClient
{
    /**
     * @var URFAClient_Connection
     */
    protected $connection = null;
    protected $address;
    protected $port;
    public $error = '';
    public function __construct($login, $pass, $address = "127.0.0.1", $port = "11758", $ssl = true, $admin)
    {
	$this->address = $address;
	$this->port = $port;
	$this->connection = new URFAClient_Connection($address, $port, $login, $pass, $ssl, $admin);
	$this->error = $this->connection->error;
    }
}
 
?>

File: lib/urfaclient.rb
require 'urfaclient_connection'
require 'urfaclient_packet'

class Urfaclient
  def initialize(login:, password:, admin:, address: "127.0.0.1", port: "11758", ssl: true)
    @address, @port = address, port
    @connection = UrfaclientConnection.new(
      login:    login,
      password: password,
      admin:    admin,
      address:  address,
      port:     port,
      ssl:      ssl
    )
    @error = @connection.error
  end
end
